### PR TITLE
feat(CDAP-20433): e2e tests for scm repo config CRUD operations

### PR DESCRIPTION
**Pull request by de-lan**
# e2e tests for scm repo config CRUD operations

TODO: add a test repository and save the config into secrets

## PR Type
- [ ] Bug Fix
- [ ] Feature
- [ ] Build Fix
- [x] Testing
- [ ] General Improvement
- [ ] Cherry Pick

## Links
Jira: [CDAP-20433](https://cdap.atlassian.net/browse/CDAP-20433)

## Screenshots
image

[CDAP-20433]: https://cdap.atlassian.net/browse/CDAP-20433?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: sandboxjs/sandbox_starter.py [hunk 1]
```diff
@@ -18,13 +18,32 @@
...
+import xml.etree.ElementTree as ET
 
 def run_shell_command(cmd):
     process = subprocess.run(cmd.split(" "), stderr=subprocess.PIPE)
     if process.returncode != 0:
         print("Process completed with error: ", process.stderr)
     assert process.returncode == 0
 
+def enable_features(path):
+  print("Enabling feature flags")
```
Comment (sumengwang) [inline sandboxjs/sandbox_starter.py]: indentation does not match above function

File: sandboxjs/sandbox_starter.py [hunk 2]
```diff
@@ -18,13 +18,32 @@
 import subprocess
```
Comment (sumengwang) [inline sandboxjs/sandbox_starter.py:18]: update copyright year

File: src/e2e-test/java/io/cdap/cdap/ui/stepsdesign/SourceControlManagement.java [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+ */
+public class SourceControlManagement {
+
+  @Then("Click on \"Link Repository\" button")
+  public void openAddRepositoryButton() {
+    ElementHelper.clickOnElement(Helper.locateElementByTestId("link-repository-button"));
+  }
+
+  @Then("Add {string} as Repository URL")
+  public void addInputLabel(String repoRul) {
```
Comment (sumengwang) [inline src/e2e-test/java/io/cdap/cdap/ui/stepsdesign/SourceControlManagement.java]: repoUrl

File: src/e2e-test/features/source.control.management.repository.feature [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+  @SOURCE_CONTROL_MANAGEMENT_TEST
+  Scenario: Should save repo config
+    When Open Source Control Management Page
+    Then Click on "Link Repository" button
+    Then Add "https://example.com" as Repository URL
+    Then Add "fake_token" as Token Name
+    Then Add "some fake token value" as Token
+    Then Click on "Save and Close" button
+    Then Verify saved repo config
```
Comment (sumengwang) [inline src/e2e-test/features/source.control.management.repository.feature:34]: It'll be better if you can verify the config table has the saved entry afterwards

File: src/e2e-test/features/source.control.management.repository.feature [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+    Then Delete the repo config
```
Comment (sumengwang) [inline src/e2e-test/features/source.control.management.repository.feature:35]: similarly verify the config table does not contain anything afterwards

File: sandboxjs/sandbox_starter.py (hunk 2 above)
Comment (de-lan) [reply sandboxjs/sandbox_starter.py:18]: updated

File: src/e2e-test/java/io/cdap/cdap/ui/stepsdesign/SourceControlManagement.java (hunk 3 above)
Comment (de-lan) [reply src/e2e-test/java/io/cdap/cdap/ui/stepsdesign/SourceControlManagement.java]: nice catch, fixed

File: sandboxjs/sandbox_starter.py (hunk 1 above)
Comment (de-lan) [reply sandboxjs/sandbox_starter.py]: updated

File: src/e2e-test/features/source.control.management.repository.feature (hunk 4 above)
Comment (de-lan) [reply src/e2e-test/features/source.control.management.repository.feature:34]: updated to verify the entries

File: src/e2e-test/features/source.control.management.repository.feature (hunk 5 above)
Comment (de-lan) [reply src/e2e-test/features/source.control.management.repository.feature:35]: Updated to verify UI is redirected back to initial page

**Review by sumengwang — APPROVED** [review]
LGTM

**Review by seanfeld — APPROVED** [review]

**Review by radhikav1 — APPROVED** [review]

**Pull request merged**